File: web/web_250423_Classic_Web/bmi_check.js
function check(){

    var name = document.main_form.name;
    var height = document.main_form.height;
    var weight = document.main_form.weight;
    var photo = document.main_form.photo;
    var check = true;

    if(input_empty_check(name)){
        alert("이름?")
        check = false;
    }

    if(input_empty_check(height) || input_onlyNum_check(height)){
        alert("키?")
        check = false;
    }
    
    if(input_empty_check(weight) || input_onlyNum_check(height)){
        alert("몸무게?")
        check = false;
    }

    if(input_empty_check(photo) || input_onlyPhoto_check(photo)){
        alert("사진?")
        check = false;
    }

    return check;
}

File: web/web_250423_Classic_Web/bmi_check.py
from uuid import uuid4
from fastapi import FastAPI, Form, UploadFile
from fastapi.responses import FileResponse, HTMLResponse


app = FastAPI()

@app.post("/bmicheck.result")
async def file_update(photo:UploadFile, name:str=Form(), height:int=Form(), weight:int=Form()):

    folder = "./photo/"
    content = await photo.read()
    type = photo.filename[-4:]
    file_name = photo.filename.replace(type, "")
    file_name = file_name + str(uuid4()) + type

    f = open(folder + file_name, "wb")
    f.write(content)
    f.close()

    h = height / 100
    bmi = weight / (h * h)

    html = "<html><head><meta charset=\"utf-8\"><title>input</title></head>"
    html += "<body>"
    html += "<h1>비만도 검사 결과</h1><hr>"
    html += "이름 : %s<br>" % name
    html += "키 : %d<br>" % height
    html += "몸무게 : %d<br>" % weight
    html += "bmi : %.1f<br>" % bmi
    html += "<img src=\"file.get?fname=%s\">" % file_name
    html += "</body></html>"

    return HTMLResponse(html)

@app.get("/file.get")
def file_get(fname):
    folder = "./photo/"
    return FileResponse(folder + fname, filename=fname)